<?xml version="1.0"?>

<launch>

  <node pkg="mpc_planner_jackalsimulator" exec="jackal_planner" name="jackal_planner" respawn="false" output="screen">
        <param from="$(find-pkg-share mpc_planner_jackalsimulator)/config/ros2_guidance_planner.yaml"/>
        <remap from="~/input/state" to="/odometry/filtered"/>
        <remap from="~/input/goal" to="/goal_pose"/>
        <remap from="~/input/reference_path" to="roadmap/reference"/>
        <remap from="~/input/obstacles" to="/pedestrian_simulator/trajectory_predictions"/>
        <remap from="~/output/command" to="/cmd_vel"/>
  </node>

  <include file="$(find-pkg-share pedestrian_simulator)/launch/ros2_simulation.launch">
    <arg name="pedestrian_scenario" value="random_social/4_corridor.xml"/>
  </include>
  <include file="$(find-pkg-share jackal_gazebo)/launch/jackal_world.launch.py"/>
  <include file="$(find-pkg-share jackal_description)/launch/description.launch.py"/>

  <include file="$(find-pkg-share roadmap)/launch/roadmap.launch">
    <arg name="map_file_name" value="maps/straight.xml"/>
  </include>

  <node pkg="tf2_ros" exec="static_transform_publisher" name="static_map" output="log" args="0 0 0 0 0 0 odom map"/>

  <node pkg="rviz2" exec="rviz2" name="rviz2" args="-d $(find-pkg-share mpc_planner_jackalsimulator)/rviz/ros2.rviz" namespace="/" output="log"/>
  
  <node pkg="rqt_reconfigure" exec="rqt_reconfigure" name="rqt_reconfigure" namespace="/jackal_planner" output="log"/>

</launch>  
